addon/lib/StudyTelemetryCollector.jsm

```diff
@@ -21,15 +22,266 @@ const BASERESOURCE = "esper-pioneer-shield-study";
 // XPCOMUtils.defineLazyModuleGetter(this, "studyUtils", STUDYUTILSPATH);
 */
 
+// telemetry utils
+const { TelemetryController } = Cu.import("resource://gre/modules/TelemetryController.jsm", null);
+const { TelemetrySession } = Cu.import("resource://gre/modules/TelemetrySession.jsm", null);
+const { TelemetryEnvironment } = Cu.import("resource://gre/modules/TelemetryEnvironment.jsm", null);
+
+XPCOMUtils.defineLazyModuleGetter(this, "PlacesUtils",
+  "resource://gre/modules/PlacesUtils.jsm");
+
 /**
  * Note: Setting studyUtils in this constructor due to above comments
  */
 class StudyTelemetryCollector {
+
   constructor(studyUtils, variation) {
     this.variation = variation;
     this.studyUtils = studyUtils;
   }
 
-  start() {
+  async start() {
+
+    this.studyUtils.telemetry({ event: "esper-init" });
+
+    // Ensure that we collect telemetry payloads only after it is fully initiated
+    Promise.all([
+      TelemetryEnvironment.onInitialized(),
+      TelemetrySession.delayedInit(),
+    ]).then(() => {
+
+      // Wait for the gather-telemetry notification before we gather telemetry
+      const gatherPromise = PromiseUtils.defer();
+      Services.obs.addObserver(gatherPromise.resolve, "gather-telemetry");
+      TelemetrySession.observe(null, "idle-daily", null);
```

Comment:
This could potentially trigger pings to be sent, messing with the Telemetry scheduling. What are you trying to achieve here?

Reply:
Same here